crates/sui-core/src/authority.rs

```diff
@@ -5101,7 +5097,10 @@ impl AuthorityState {
         let tx = EndOfEpochTransactionKind::new_store_execution_time_observations(
             epoch_store
                 .get_end_of_epoch_execution_time_observations()
-                .filter_and_sort_v1(|(key, _)| included_execution_time_observations.contains(key)),
+                .filter_and_sort_v1(
+                    |(key, _)| included_execution_time_observations.contains(key),
+                    params.stored_observations_limit.try_into().unwrap(),
```

Comment:
why not just make the function take u64 and get rid of the unwrap?

Reply:
It has to go somewhere, no? to me it seemed more natural for `filter_and_sort_v1`, which doesn't depend on protocol config, to take the type it needs, and have the function that does manage the protocol config do the conversion